internal/component_model.go

```diff
@@ -111,12 +117,85 @@ func (m *componentResourceModel) toRemoteInput() sbmgmt.ComponentInput {
 	// using the ordering of the json...
 	schema := sortComponentFields(raw)
 
-	return sbmgmt.ComponentInput{
-		Name:               m.Name.ValueString(),
-		Schema:             schema,
-		IsNestable:         m.IsNestable.ValueBoolPointer(),
-		IsRoot:             m.IsRoot.ValueBoolPointer(),
-		ComponentGroupUuid: ref(asUUID(m.ComponentGroupUUID)),
+	componentGroupUuid := uuid.Must(uuid.FromString(m.ComponentGroupUUID.ValueString()))
+
+	return sbmgmt.CreateComponentJSONRequestBody{
+		Component: &sbmgmt.ComponentCreateInput{
+			Color:              m.Color.ValueStringPointer(),
+			ComponentGroupUuid: &componentGroupUuid,
+			DisplayName:        m.DisplayName.ValueStringPointer(),
+			Icon:               (*sbmgmt.ComponentCreateInputIcon)(m.Icon.ValueStringPointer()),
+			Image:              m.Image.ValueStringPointer(),
+			IsNestable:         m.IsNestable.ValueBoolPointer(),
+			IsRoot:             m.IsRoot.ValueBoolPointer(),
+			Name:               m.Name.ValueString(),
+			Preview:            m.Preview.ValueStringPointer(),
+			Schema:             schema,
+		},
+	}
+}
+func (m *componentResourceModel) toUpdateInput() sbmgmt.UpdateComponentJSONRequestBody {
+
+	raw := make(map[string]sbmgmt.FieldInput, len(m.Schema))
+	for name := range m.Schema {
+		item := m.Schema[name]
+		raw[name] = sbmgmt.FieldInput{
```

Comment:
We could put some of this in a function so that both the update and create have a shared list of items.